### PR TITLE
add more projection pushdown slt tests

### PR DESCRIPTION
**Pull request by adriangb**
Adds more interesting test cases extracted from https://github.com/apache/datafusion/pull/19538

The reasoning is that it's (1) better to split the additional tests out from that PR to make that PR easier to review and (2) it's a good thing to document current behavior with complex edge cases like these, so even if that PR doesn't get merged there is still value in having these as regression tests.

### CODE REVIEW

File: datafusion/sqllogictest/test_files/projection_pushdown.slt [hunk 1]
```diff
@@ -1038,7 +1084,295 @@ SELECT id, s['value'] + 100, s['label'] || '_test' FROM simple_struct ORDER BY i
...
+EXPLAIN SELECT s['value'], s['label'], id FROM simple_struct WHERE id > 1;
+----
+logical_plan
+01)Projection: get_field(simple_struct.s, Utf8("value")), get_field(simple_struct.s, Utf8("label")), simple_struct.id
+02)--Filter: simple_struct.id > Int64(1)
+03)----TableScan: simple_struct projection=[id, s], partial_filters=[simple_struct.id > Int64(1)]
+physical_plan
+01)ProjectionExec: expr=[get_field(s@1, value) as simple_struct.s[value], get_field(s@1, label) as simple_struct.s[label], id@0 as id]
+02)--FilterExec: id@0 > 1
+03)----DataSourceExec: file_groups={1 group: [[WORKSPACE_ROOT/datafusion/sqllogictest/test_files/scratch/projection_pushdown/simple.parquet]]}, projection=[id, s], file_type=parquet, predicate=id@0 > 1, pruning_predicate=id_null_count@1 != row_count@2 AND id_max@0 > 1, required_guarantees=[]
```
Comment (Jefffrey) [inline datafusion/sqllogictest/test_files/projection_pushdown.slt]: Could you help explain this test to me? Comments state

> Physical plan shows: FilterExec: id@2 > 1 (note the index after reordering).

But the plan shows

> 02)--FilterExec: id@0 > 1

Is this expected?

Comment (adriangb) [reply datafusion/sqllogictest/test_files/projection_pushdown.slt]: Probably should just remove the comment. These were pulled out of another PR and updated, the index will change once the `ProjectionExec` is pushed through the `FilterExec`.

Comment (adriangb) [reply datafusion/sqllogictest/test_files/projection_pushdown.slt]: Amended in 37e75b1

File: datafusion/sqllogictest/test_files/projection_pushdown.slt [hunk 2]
```diff
@@ -361,13 +361,58 @@ SELECT id, s['value'] FROM simple_struct ORDER BY s['value'];
 5 250
 4 300
 
+###
+# Test 4.4: Projection with duplicate column through Sort
+# The projection expands the logical output (3→4 columns) but reduces physical columns
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/projection_pushdown.slt]: I don't understand the comment "reduces physical columns" as there are three physical column and all three are scanned 🤔

File: datafusion/sqllogictest/test_files/projection_pushdown.slt [hunk 3]
```diff
@@ -361,13 +361,58 @@ SELECT id, s['value'] FROM simple_struct ORDER BY s['value'];
...
+01)Sort: three_cols.col_a ASC NULLS LAST
+02)--Projection: three_cols.col_a, three_cols.col_b, three_cols.col_c, three_cols.col_b AS col_b_dup
+03)----TableScan: three_cols projection=[col_a, col_b, col_c]
+physical_plan
+01)SortExec: expr=[col_a@0 ASC NULLS LAST], preserve_partitioning=[false]
+02)--DataSourceExec: file_groups={1 group: [[WORKSPACE_ROOT/datafusion/sqllogictest/test_files/scratch/projection_pushdown/three_cols.parquet]]}, projection=[col_a, col_b, col_c, col_b@1 as col_b_dup], file_type=parquet
+
+# Verify correctness
+query IIII
+SELECT col_a, col_b, col_c, col_b as col_b_dup FROM three_cols ORDER BY col_a;
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/projection_pushdown.slt]: given this data was inserted in order of `col_a` it might be better to check a different sort order (e.g. sort by col_a desc)

File: datafusion/sqllogictest/test_files/projection_pushdown.slt (hunk 2 above)
Comment (adriangb) [reply datafusion/sqllogictest/test_files/projection_pushdown.slt]: The point is that after the projection there are 4 columns but before the projection there are 3 columns. But I agree the comment is confusing, I'll reword it
